### ISSUE TITLE
Reading Custom Log Quantity from GSD file

### ISSUE THREAD

**Issue by tommy-waltmann** [glotzerlab/hoomd-blue#1336]
## Description

It appears that I cannot read custom log quantities from a gsd file, because they are not being written to the gsd.

## Script

To run the simulation ...
```python
import hoomd


def make_snapshot():
    snap = hoomd.Snapshot()
    snap.particles.N = 1
    snap.particles.position[:] = [[0, 0, 0]]
    snap.particles.types = ['A']
    snap.particles.typeid[:] = [0]
    return snap


class LogSingleValue(hoomd.custom.Action):

    def __init__(self):
        self._value = 5

    @hoomd.logging.log
    def value(self):
        return self._value

    def act(self):
        pass


dev = hoomd.device.CPU()
sim = hoomd.Simulation(dev)
sim.create_state_from_snapshot(make_snapshot())

value_action = LogSingleValue()
log = hoomd.logging.Logger()
log.add(value_action, quantities=['value'])

gsd_writer = hoomd.write.GSD(trigger=hoomd.trigger.Periodic(1),
                             filename='sim.gsd',
                             mode='wb',
                             log=log)
sim.operations.add(gsd_writer)

sim.run(2)
```

To try and read from the gsd ...
``` python
import gsd.hoomd

traj = gsd.hoomd.open('sim.gsd')
frame = traj[0]
value = frame.log['LogSingleValue/value']
print(value)
```

## Output

``` bash
Traceback (most recent call last):
  File "read_gsd.py", line 5, in <module>
    value = frame.log['LogSingleValue/value']
KeyError: 'LogSingleValue/value'
```

## Expected output

``` bash
5
```

## Configuration

Platform:
- Mac
- CPU

Installation method:
- Compiled from source

Versions:
- Python version: 3.7.12
- **HOOMD-blue** version: 3.2.0
- gsd version: 2.4.2

## Developer

I could have a look at this when I get the time, but @b-butler implemented the custom actions and logging subsystem, so he would probably be able to fix it faster than I could.

**Comment by b-butler** [comment]
@tommy-waltmann can you give the output of `list(logger)`

**Comment by joaander** [comment]
The value is written. It adds the `__main__` namespace:
```
>>> import gsd.hoomd
>>> traj = gsd.hoomd.open('sim.gsd')
>>> frame = traj[0]
>>> frame.log
{'__main__/LogSingleValue/value': array([5])}
```
Is this intended behavior? Is it consistent with earlier releases?

I thought we removed namespaces when they didn't start with `hoomd`.

**Comment by tommy-waltmann** [comment]
output of `list(logger)` is
``` bash
[('__main__', 'LogSingleValue', 'value'), ('__main__', 'LogSingleValue'), ('__main__',)]
```

**Comment by b-butler** [comment]
Hmm, that reveals an unexpected bug in `__iter__`.

@joaander custom actions are first class citizens in a sense. They allow for automatic namespacing. We only remove selected namespaces in logging internally (e.g. hoomd); we never remove a namespace from a user's code (we could if they used the exact module hierarchy we do now that I am thinking about it).

We could remove `__main__` to prevent that from showing up (`__main__` is the module name given to the script being run).

**Comment by joaander** [comment]
We could either remove `__main__` or document the behavior. I can see how this would be useful if you had your custom action in a package:
`['my_package/LogSingleValue/value', 'someone_elses_package/LogSomethingElse/value']`

**Comment by joaander** [comment]
Fixed in #1337.
